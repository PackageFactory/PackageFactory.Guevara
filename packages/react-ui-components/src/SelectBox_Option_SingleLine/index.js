/* eslint-disable camelcase, react/jsx-pascal-case */
import SelectBox_Option_SingleLine from './selectBox_Option_SingleLine.js';

export default SelectBox_Option_SingleLine;
