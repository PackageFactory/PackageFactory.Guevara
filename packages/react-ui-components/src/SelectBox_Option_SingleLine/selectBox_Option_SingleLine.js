/* eslint-disable camelcase, react/jsx-pascal-case */
import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import ListPreviewElement from '../ListPreviewElement/index';
import mergeClassNames from 'classnames';

export default class SelectBox_Option_SingleLine extends PureComponent {
    static propTypes = {
        option: PropTypes.shape({
            label: PropTypes.string.isRequired,
            icon: PropTypes.string,
            disabled: PropTypes.bool
        }).isRequired,

        disabled: PropTypes.bool,

        className: PropTypes.string
    };

    render() {
        const {option, className, disabled} = this.props;

        const finalClassNames = mergeClassNames({
            [className]: className
        });

        return (
            <ListPreviewElement {...this.props} icon={option.icon} disabled={disabled} className={finalClassNames}>
                <span>{option.label}</span>
            </ListPreviewElement>
        );
    }
}
