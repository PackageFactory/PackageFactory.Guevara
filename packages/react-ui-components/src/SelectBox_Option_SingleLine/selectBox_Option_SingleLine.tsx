// tslint:disable:class-name
import mergeClassNames from 'classnames';
import React, {PureComponent} from 'react';

import ListPreviewElement from '../ListPreviewElement';
import {ListPreviewElementProps} from '../ListPreviewElement/listPreviewElement';
import {SelectBoxOption} from '../SelectBox/selectBox';

export interface SelectBox_Option_SingleLineProps extends Partial<ListPreviewElementProps> {
    readonly option: SelectBoxOption;
    readonly disabled?: boolean;
    readonly className?: string;
}

export default class SelectBox_Option_SingleLine extends PureComponent<SelectBox_Option_SingleLineProps> {
    public render(): JSX.Element {
        const {option, className, disabled, ...rest} = this.props;

        const isDisabled = disabled || option.disabled;

        const finalClassNames = mergeClassNames(className);

        return (
            <ListPreviewElement {...rest} icon={option.icon} disabled={isDisabled} className={finalClassNames}>
                <span>{option.label}</span>
            </ListPreviewElement>
        );
    }
}
