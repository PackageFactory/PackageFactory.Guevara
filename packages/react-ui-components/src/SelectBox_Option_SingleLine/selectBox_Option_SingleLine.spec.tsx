/* eslint-disable camelcase, react/jsx-pascal-case */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import SelectBox_Option_SingleLine, {SelectBox_Option_SingleLineProps} from './selectBox_Option_SingleLine';

describe('<SelectBox_Option_SingleLine/>', () => {
    const props: SelectBox_Option_SingleLineProps = {
        children: 'Foo children',
        onClick: jest.fn(),
        option: {
            label: 'Foo label',
            icon: 'level-up',
        },
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<SelectBox_Option_SingleLine {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
