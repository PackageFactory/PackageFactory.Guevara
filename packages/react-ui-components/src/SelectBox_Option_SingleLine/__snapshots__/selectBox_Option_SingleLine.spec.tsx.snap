// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectBox_Option_SingleLine/> should render correctly. 1`] = `
<ThemedListPreviewElement
  className=""
  composeTheme="deeply"
  icon="level-up"
  isHighlighted={false}
  mapThemrProps={[Function]}
  onClick={[MockFunction]}
>
  <span>
    Foo label
  </span>
</ThemedListPreviewElement>
`;
